prime-router/src/main/kotlin/azure/ConditionMapper.kt

```diff
@@ -0,0 +1,138 @@
+package gov.cdc.prime.router.azure
+
+import gov.cdc.prime.router.ActionLogDetail
+import gov.cdc.prime.router.Metadata
+import gov.cdc.prime.router.UnmappableConditionMessage
+import gov.cdc.prime.router.fhirengine.utils.getCodeSourcesMap
+import gov.cdc.prime.router.fhirengine.utils.getObservations
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Observation
+
+interface ConditionMapper {
+    /**
+     * Attempt to find diagnostic conditions for a given test [code]
+     * @return a list of diagnostic conditions identified (could be empty)
+     */
+    fun lookupCondition(code: String): List<Coding>
+
+    /**
+     * Attempt to find diagnostic conditions for a series of test [codes]
+     * @return a map associating test [codes] to their diagnostic conditions as Coding's
+     */
+    fun lookupConditions(codes: List<String>): MutableMap<String, List<Coding>>
+
+    /**
+     * Lookup condition codes on every observation in a [bundle] and add them as custom extensions
+     * @param bundle the bundle that will be stamped
+     * @return a list of ObservationMappingFailure objects with information on any mapping failures and their source
+     */
+    fun stampBundle(bundle: Bundle): List<LookupTableConditionMapper.ObservationMappingFailure>
+
+    /**
+     * Lookup condition codes for an [observation] and add them as custom extensions
+     * @param observation the observation that will be stamped
+     * @return a list of ActionLogDetail objects with information on any mapping failures
+     */
+    fun stampObservation(observation: Observation): List<ActionLogDetail>
```

Comment:
Might make more sense for this to accept the observation and codes to add as an extension?  Feels like making the interface return an ActionLogDetail is limiting.

Reply:
On second thought this probably doesn't belong in this interface whatsoever. More to come.